drivers/azure/azureutil/azureutil.go

```diff
@@ -520,6 +534,32 @@ func (a AzureClient) CreateVirtualMachine(resourceGroup, name, location, size, a
 	return err
 }
 
+func (a AzureClient) CreateVirtualMachineExtension(os, resourceGroup, name, location string) error {
+	vmExtension := compute.VirtualMachineExtension{
```

Comment:
Hmm I suggest moving it up here. B/c if say the next statement gets a nil dereference panic, we'll know where it comes from very clearly.